### ISSUE TITLE
CA1801 false positive on record parameter if any parameter has a default value

### ISSUE THREAD

**Issue by thomaslevesque** [dotnet/roslyn-analyzers#4462]
### Analyzer

**Diagnostic ID**: [CA1801](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1801): `Review unused parameters`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-rc.2](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

If any parameter of a positional record has a default value, CA1801 is reported on all parameters.

### Steps To Reproduce

Write this code:

```csharp
public record Foo(string Name, int Age = 0);
```

### Expected behavior

No warning

### Actual behavior

CA1801 is reported on all parameters:

> Parameter Name of method .ctor is never used. Remove the parameter or use it in the method body.
> Parameter Age of method .ctor is never used. Remove the parameter or use it in the method body.

## Additional context

Looks very similar to #4397

**Comment by Evangelink** [comment]
I confirm that I can reproduce the issue.

Regarding how to fix the issue, I am not sure what to do. I think there is a bug (or at least something missing) on roslyn because the property `IsImplicitlyDeclared` returns false while IMO it should return true.

@mavasani let me know if I am missing something.

**Comment by Youssef1313** [comment]
@Evangelink See similar discussion in https://github.com/dotnet/roslyn/issues/48193

**Comment by fjallemark** [comment]
Is this same issue? I get CA1801 on parameter *AssignedRoute* i.e. when "subclassing" record and add properties on the sub-record.

```
public abstract record InputEvent(long Id, DateTimeOffset Timestamp) : MovableEvent(Id, Timestamp);
public sealed record AssignRouteEvent(long Id, DateTimeOffset Timestamp, Route AssignedRoute) : InputEvent(Id, Timestamp);
```

**Comment by Evangelink** [comment]
@fjallemark For me the issues look the same.
